File: app/src/main/java/br/com/jonathanzanella/myexpenses/sync/UnsyncModelApi.java
package br.com.jonathanzanella.myexpenses.sync;

import java.util.List;

public interface UnsyncModelApi<T extends UnsyncModel> {
	List<T> index();
	void save(UnsyncModel model);
	void syncAndSave(UnsyncModel unsync);
	List<T> unsyncModels();
	long greaterUpdatedAt();
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/sync/UnsyncModelApi.kt
package br.com.jonathanzanella.myexpenses.sync

interface UnsyncModelApi<out T : UnsyncModel> {
    fun index(): List<T>
    fun save(model: UnsyncModel)
    fun syncAndSave(unsync: UnsyncModel)
    fun unsyncModels(): List<T>
    fun greaterUpdatedAt(): Long
}